modyn/selector/internal/selector_strategies/downsampling_strategies/abstract_downsampling_strategy.py

```diff
@@ -42,19 +42,11 @@ def __init__(
         self.requires_remote_computation = True
         self.maximum_keys_in_memory = maximum_keys_in_memory
         self.downsampling_config = downsampling_config
-        self.status_bar_scale = self._compute_status_bar_scale()
-
-    def _compute_status_bar_scale(self) -> int:
-        """
-        This function is used to create the downsampling status bar and handle the training one accordingly.
-
-        For BTS, we return 100 since the training status bar sees all the samples
-        For STB, we return the downsampling_ratio since the training status bar sees only a fraction of points
-        (while the downsampling status bas sees all the points)
-        """
-        if self.downsampling_mode == DownsamplingMode.BATCH_THEN_SAMPLE:
-            return 100
-        return self.downsampling_ratio
+        # the status bar scale is used in conjunction with the total number of samples (after presampling)
+        # and the number of currently trained batches to show current training progress
+        # No matter it is BtS or StB, the number of trained batches should be compared to the total number of samples
+        # divided by the downsampling ratio. Therefore, the status bar scale should be the downsampling ratio.
+        self.status_bar_scale = self.downsampling_ratio
```

Comment:
There are a bit more changes involved @MaxiBoether , as I found the visual status bar on client side could still be inaccurate. It's in a confusing state but I propose to solve it ultimately with the following approach:

The client gets 4 numbers:
1. the total number of samples after presampling
2. status bar scale
3. the number of seen samples for StB downsampling
4. the number of *trained* samples

On client side, [1, 3] are compared to show StB downsampling progress. [1, 2, 4] are used to show training progress by comparing `total_num_samples / status_bar_scale` with `num_trained_samples`. Now no matter what mode it is in, it is always correct.

What also is changed is in `pytorch_trainer`, the number fed into `update_queue` now are the number of trained batches and the number of trained samples.

Reply:
I think this sounds reasonable. If I understand correctly, for training, you basically want to show `number of trained samples / number of samples we will train on` - which I think is the best solution!

Reply:
yes exactly!